src/main/java/com/synopsys/integration/alert/web/api/custom/CustomEndpointController.java

```diff
@@ -66,7 +70,18 @@ public CustomEndpointController(CustomEndpointManager customEndpointManager, Res
         }
 
         HttpServletContentWrapper servletContentWrapper = new HttpServletContentWrapper(httpRequest, httpResponse);
-        return customEndpointManager.performFunction(key, restModel, servletContentWrapper);
+        ActionResult<?> result = customEndpointManager.performFunction(key, restModel, servletContentWrapper);
+        if (result.isOk()) {
+            if (result.hasContent()) {
+                String content = gson.toJson(result.getContent().get());
+                return responseFactory.createOkContentResponse(content);
+            } else {
+                String message = result.getMessage().orElse("");
+                return responseFactory.createOkResponse("", message);
+            }
+        } else {
+            String message = result.getMessage().orElse("");
+            return responseFactory.createResponse(result.getHttpStatus(), null, message);
+        }
```

Comment:
If we add the conversion to ActionResult, this could all be handled in a methodin ActionResult.